### ISSUE TITLE
Getting invalid prop-type warnings when destructing

### ISSUE THREAD

**Issue by redonkulus** [yannickcr/eslint-plugin-react#2327]
We have a bunch of functional components that use destructuring to obtain their props, in version 7.14.0, we started getting errors like this:

```
/home/foo/src/components/Container/YourSections.jsx
  14:18  error  'sections.length' is missing in props validation  react/prop-types
  85:40  error  'sections.map' is missing in props validation     react/prop-types
```

based on this code:

```js
const YourSections = ({ sections = [] }) => {
    if (sections.length < 1) {
        return;
    }
    // rest of code removed
};
```

eslint config snippet:

```js
module.exports = {
    // ...
    rules: {
        // ...
        'react/display-name': 1,
        'react/jsx-key': 1,
        'react/jsx-no-bind': 1,
        'react/jsx-no-comment-textnodes': 2,
        'react/jsx-no-duplicate-props': 2,
        'react/jsx-no-target-blank': 1,
        'react/jsx-no-undef': 2,
        'react/jsx-uses-react': 2,
        'react/jsx-uses-vars': 2,
        'react/no-deprecated': 2,
        'react/no-string-refs': 1,
        'react/prop-types': 1,
        'react/react-in-jsx-scope': 2
        // ...
    }
};
```

We tried downgrading to version 7.13.0 and the fixed the issue. So it seems to be related to one of the PRs from 7.14.0 that handles nested destructuring.

eslint version: 5.16.0
eslint-plugin-react: 7.14.2

**Comment by ljharb** [comment]
Add:
```jsx
YourSections.propTypes = {
  sections: PropTypes.array,
};
```
and both the warning will go away, and your component will be correctly propTyped.

**Comment by redonkulus** [comment]
@ljharb my mistake, my code snippet didnt include the prop-types code, which my local did. However, it turns out I had a mistake in my prop types declaration which was exposed in 7.14.0. So I should say thank you then 😄.

Will close this.
